### PR TITLE
Fix some documentation omissions

### PR DESCRIPTION
**Pull request by starwed**
Just a couple of changes to documentation:
- The callback for .onHit()
- The synchronous nature of Crafty events

The first came up in the forum, and the second is something that confused me when I was learning crafty.  (I'd at first assumed it worked like a DOM event.)

### CODE REVIEW

**Comment by starwed** [comment]
Ok, I'm learning git and confused about how to squash commits, or something -- let me try again in a bit!  :)

**Comment by mpetrovic** [comment]
Looks like there's a bit more than documentation changes. Could you review those?

**Comment by starwed** [comment]
Ok, should be just the doc changes now.  I got into trouble trying to squash some commits together.  Now I know a lot more about how to use rebase and cherry-pick... :)

**Comment by mpetrovic** [comment]
its still got quite a few extra

**Comment by starwed** [comment]
Hmm, I'm confused then -- when I look at this with the github interface, I see only the two changes to documentation I describe.  

The only diffs I see affect `src/collision.js` on lines 155 and 158, and `src/core.js` on lines 491 and 578.  What extras are you seeing?  

(That's a total of 2 commits, and 2 files affected.  My earlier rebase error produced something like 8 commits total.)

**Comment by mpetrovic** [comment]
It changed still since I looked 15 minutes ago. All green here.
